e2e/features/Udacity.feature

```diff
@@ -14,4 +14,20 @@ Scenario: 1. The user should be able to use the search bar
    Then the course count should be less than the remembered course count
 
    When the search bar is cleared
-   Then the course count should be equal to the remembered course count
\ No newline at end of file
+   Then the course count should be equal to the remembered course count
+
+Scenario: 2. The card details and the filter fields should work correctly
+  Given the Udacity course page is opened
+    
+   When the text "javascript" is typed into the search bar
+   Then the selected filters field should contain the "javascript" text
+    And the course count should be equal to the number of the courses that visible
+
+Scenario: 3. The card details, dropdown filter and filter fields should work correctly
+  Given the Udacity course page is opened
+
+   When the "Select Program Details" dropdown is clicked
+    And the "Intermediate" skill level is selected
+   Then the selected filters field should contain the "Intermediate" text
+    And the course count should be equal to the number of the courses that visible
+    And the course level logo and the "Intermediate" label should be visible
```

Comment:
Please, create two sepaarte steps for these assertions, one for label and one for the icon.